### PR TITLE
Add recipe for pomidor

### PR DESCRIPTION
**Pull request by TatriX**
### Brief summary of what the package does

Pomidor is a simple pomodoro timer.

### Direct link to the package repository

https://github.com/TatriX/pomidor

### Your association with the package
I'm a maintainer.

### Relevant communications with the upstream package maintainer
**None needed**

### Checklist

- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] I've built and installed the package using the instructions in the [README](https://github.com/melpa/melpa/blob/master/README.md)

### CODE REVIEW

File: recipes/pomidor [hunk 1]
```diff
@@ -0,0 +1 @@
+(pomidor :fetcher github :repo "TatriX/pomidor" :files ("pomidor.el" "tick.wav" "tack.wav" "overwork.wav"))
```
Comment (syohex) [inline recipes/pomidor]: Please use `:defaults` instead of `pomidor.el`.

**Comment by syohex** [comment]
I have sent PR about packaging issue. Please check it(https://github.com/TatriX/pomidor/pull/2)

**Comment by purcell** [comment]
Thanks. Quick feedback from me:
- There's no standard "productivity" custom group. Please choose a different parent, e.g. `applications` or `environment`.
- Prefer `defconst` to `defvar` for values like `pomidor-dir` etc.
- Don't incude a font or width in `pomidor-time-face`: those are very personal and platform-specific settings. Also, don't set `:foreground` unless you are going to provide variants for `dark` and `light` backgrounds. Consider just `:inherit`-ing standard faces like `warning`, `success`, `font-lock-comment-face` by default, so that themes can easily work.
- Use the `async` package to play the sound asynchronously: it will handle creating a child process for you.
- You should pass the `LOCAL` flag to `(add-hook 'kill-buffer-hook ...), otherwise you'll be cancelling the timer when _any_ buffer is killed.

Hope that helps!

**Comment by purcell** [comment]
Also, there was an excellent suggestion on Reddit that you use `alert` to notify the user: this is definitely the way to go. I'm not sure why `(alert "test" :style 'libnotify)` didn't work for you, but you generally shouldn't hard-code the style: `(alert "test")` should do the right thing on all platforms, and if you provide `:severity` and/or `:category`, users can apply their own preferences to control how the notifications appear.

**Comment by TatriX** [comment]
@purcell thank you. I've pushed some changes based on your feedback.

Where can I find info about `async`?

How do  you think `alert` should be used from the user side? e.g. how can I disable notification? How can I adjust what and how is shown?

**Comment by TatriX** [comment]
So I should call from my side:
```list
(alert "Make a pause" :severity 'high :category 'pomidor)
```
This way users can control appearance or for example ignore the notification. But how should I make the message customizable? Probably I should add a function which will be called to produce a message. Then user will be able to override it.

**Comment by purcell** [comment]
> This way users can control appearance or for example ignore the notification.

Yes, exactly. 

> But how should I make the message customizable? Probably I should add a function which will be called to produce a message. Then user will be able to override it.

Maybe use a `defcustom` to define a variable for each message?

Regarding `async` and the sounds, I think you would just use one of the `async` functions to call `play-sound-file` asynchronously. Check out the README.

**Comment by TatriX** [comment]
Hm, I've tried async and it seems it's worse then current implementation.
It has problems with passing params:
```lisp
(async-start (lambda () (play-sound-file pomidor-sound-overwork)) 'ignore)
;; error in process sentinel: Symbol’s value as variable is void: pomidor-sound-overwork
```
Then it's a dependency, when actually I need only 3 lines of code.
Maybe I'm doing something wrong?

**Comment by purcell** [comment]
Yeah, so that lambda will run in a separate process, where `pomidor-sound-overwork` will not be defined. You'd need to use [async-inject-variables](https://github.com/jwiegley/emacs-async#async-inject-variables). If you can get it to work, it's still better than rolling your own sub-process execution code IMO.

**Comment by TatriX** [comment]
I read async-start code and it's almost identical to my own solution (for my "play audio file" case).
So I think making a dependency on the library to make such a simple thing isn't the best idea. Less dependencies — less complexity. IMO ;)

What else should be resolved to merge this PR?

File: recipes/pomidor (hunk 1 above)
Comment (TatriX) [inline recipes/pomidor]: It's done, but I think github already told you so.

**Comment by syohex** [comment]
Merged

**Pull request merged**